packages/model-serving/src/components/deploymentWizard/ModelDeploymentWizard.tsx

```diff
@@ -105,24 +105,36 @@ const ModelDeploymentWizard: React.FC<ModelDeploymentWizardProps> = ({
     <ApplicationsPage title={title} description={description} loaded empty={false}>
       <Wizard onClose={exitWizard} onSave={onSave} footer={<WizardFooterWithDisablingNext />}>
         <WizardStep name="Source model" id="source-model-step">
-          <ModelSourceStepContent wizardState={wizardState} validation={validation.modelSource} />
+          {wizardState.loaded.modelSourceLoaded ? (
+            <ModelSourceStepContent wizardState={wizardState} validation={validation.modelSource} />
+          ) : (
+            <Spinner />
+          )}
```

Comment:
It would be nice if there was a way to pass a loading state to the `WizardStep`...

Reply:
eh alternatively we could do this inside each step content. Maybe that makes more sense actually